cpp/src/table/row_operators.cu

```diff
@@ -28,6 +30,52 @@ namespace experimental {
 
 namespace {
 
+/**
+ * @brief Applies the offsets of struct columns onto their children.
+ *
+ * Given a table, this replaces any struct columns with similar struct columns that have their
+ * offsets applied to their children. Structs that are children of list columns are not affected.
+ *
+ */
+table_view pushdown_struct_offsets(table_view table)
+{
+  std::function<std::vector<column_view>(column_view const&)> slice_children =
```

Comment:
Maybe this matters, maybe not, but it seems like it'd be just as easy to pull this into a normal, named function and avoid the `std::function` overhead. I'm guessing that gets in the way of any optimization the compiler can do on the recursion.